### PR TITLE
Update Version Numbers

### PR DESCRIPTION
**Pull request by KevinRansom**
Here is the update to FSharp.Core 4.4.3.

This updates the version numbers of everything :-(

1.  The compiler and related assemblies go from 4.4.1.0   --->  4.4.3.0   // We skipped 4.4.2.0 to align with nuget package
2.  The VS Assemblies go from 15.4.1.0 ---> 15.4.3.0
3.  The Nuget package goes from 4.2.x ---> 4.3.x
4.  The project properties Target F# selection is updated.
5.  Updated some tests to use the parameterized portable libraries nuget package.

image

### CODE REVIEW

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@cartermp think of 4.2 as an internal version that never shipped.

**Comment by cartermp** [comment]
I get that, but the point is that we shouldn't surface these messy version numbers in the UI like this.

In some future, I'll be using the F# compiler to compile F# 4.2 code. But in the project settings, I'll need to use "F# 4.3" or higher as my target.  That makes no sense at all.

**Comment by KevinRansom** [comment]
You need to select an FSharp.Core version.  You know it is a nuget packat and you want F# 4.4.2.0,  so you know that you should use the nuget package F# 4.2 ....  

Duh!!!! no you can't do that because we already used 4.2 as a nuget version number, and it contains F# 4.4.1.0.  We had to do that, odious as it was, because 4.1 was already used by the community package, and we cut down hugely on the platforms shipped with the nuget package., a likely breaking change
and we also weren't ready to rev the FSharp.Core version number.

Skipping 4.2 for assembly version numbers allows us to get back on track ... sort of.

From here on forward, the Major Minor of the Nuget package will match the SDK version.  4.3 ... 4.4 ... 4.5 etc.

4.1 will even be okay since it has a recently published FSharp.Core  it's only 4.2 that's iffy, because it also contains FSharp.Core 4.4.1.0.

Anyway that is my 2 cents and worth almost every penny

Kevin

**Comment by isaacabraham** [comment]
My two cents (whether wanted or not...) is that this sounds like a complete mess. I don't understand why this is mismatched or what the reasoning behind it. It just seems really, really complicated and I don't understand why :-/

**Comment by cartermp** [comment]
I don't particularly mind what the version numbers are.  It's already hopelessly confusing, so getting it back on track is important.  I think you're missing my point @KevinRansom.  The UI in your screenshot says "F# 4.3".  F# 4.3 doesn't even exist, because F# 4.2 doesn't even exist.  I believe that this will confuse people _further_ than they are already.

**Comment by KevinRansom** [comment]
@cartermp but 4.2 doesn't exist so adding it to the drop down seems more confusing.  Anyway, the drop down is about selecting the FSharp.Core version number.  And no 4 version 4.4.2.0.  I am hopefull that some future version of the drop down will use nuget packaging rather than raw FSharp.Core version numbers.

I believe this dialog was designed for VS 2013 and so didn't really take nuget into account.

We can do better in the future.

**Comment by cartermp** [comment]
I'm saying that this string:

"F# 4.3"

Will not be interpreted as "the version of FSharp.Core that succeeds the first `4.`".  People will think that this means the _language version_.  If possible, this needs to be cleaned up.

**Comment by KevinRansom** [comment]
I welcome suggestions.

**Comment by KevinRansom** [comment]
@cartermp I didn't design the dialog, and feel no need to defend it.

In this context F# 4.3 means the F# 4.3 SDK, which is located at: %ProgramFiles(x86)%\Microsoft\SDKs\F#\4.3\**\**

The branded version number displayed in FSI and FSC is not the language version number or even the Compiler or interactive version number, it's more like the SDK version.

Kevin

**Comment by cartermp** [comment]
Okay, so @KevinRansom and I chatted about this and the issue that I raised is ultimately orthagonal and an issue in the tooling.  The version number (F# 4.3) is **not** the language version, but the Compiler SDK version.  This is what it's always been, and it's horribly confusing.  The long-term solution is to tie all of that into a NuGet package that the tooling snags, but that won't be something we do immediately.

I created #4070 to track this.

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@dotnet-bot test Ubuntu14.04 Release_fcs Build please

**Comment by dsyme** [comment]
Guys, can we **please**, **please** not use “F# 4.3” in the banner text for fsc.exe and fsi.exe?


> The version number (F# 4.3) is not the language version, but the Compiler SDK version. This is what it's always been, and it's horribly confusing. The long-term solution is to tie all of that into a NuGet package that the tooling snags, but that won't be something we do immediately.

This is not the case. The banner text  has **always** been the F# language version historically, right from F# 0.4 or something. 

As of today, there is no such thing as “F# 4.3” or “Microsoft Visual F# 4.3” or any of that.  Nearly all the updates in this PR are thus plain wrong. Indeed let’s just minimize the use of the text “F# 4.3”.  

I don’t care what you use for the “Microsoft SDKs” folder.  But please don’t use the SDK version number in the banner except in parentheses or something.

File: setup/FSharp.SDK/FSharp.SDK.wxs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <!-- Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information. -->
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
 
-  <?define ProductVersion = "4.1"?>
+  <?define ProductVersion = "4.3"?>
```
Comment (dsyme) [inline setup/FSharp.SDK/FSharp.SDK.wxs]: Why don't we use "F# 4.1.3" here?  It is really, really confusing to be adding a new series of SDK version numbers which are so immensely confusing.

File: setup/resources/Microsoft.Portable.FSharp.Shim.targets [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   
-  <Import Project="$(MSBuildProgramFiles32)\Microsoft SDKs\F#\4.1\Framework\v4.0\Microsoft.Portable.FSharp.Targets" />
+  <Import Project="$(MSBuildProgramFiles32)\Microsoft SDKs\F#\4.3\Framework\v4.0\Microsoft.Portable.FSharp.Targets" />
```
Comment (dsyme) [inline setup/resources/Microsoft.Portable.FSharp.Shim.targets]: Again why not ``4.1.3`` here?

File: src/fsharp/FSharp.Core.nuget/FSharp.Core.LatestNuget.nuspec [hunk 3]
```diff
@@ -3,10 +3,10 @@
     <metadata>
         <!-- rename to FSharp.Core when done -->
         <id>FSharp.Core</id>
-        <title>FSharp.Core for F# 4.1</title>
-        <summary>FSharp.Core for F# 4.1</summary>
+        <title>FSharp.Core for F# 4.3</title>
```
Comment (dsyme) [inline src/fsharp/FSharp.Core.nuget/FSharp.Core.LatestNuget.nuspec]: This update is badly wrong.  It should be "FSharp.Core for F# 4.1" or "FSharp.Core 4.4.3.0".

File: src/fsharp/FSharp.Core.nuget/FSharp.Core.LatestNuget.nuspec [hunk 4]
```diff
@@ -3,10 +3,10 @@
...
+        <summary>FSharp.Core for F# 4.3</summary>
         <description>
-            FSharp.Core redistributables from Visual F# 4.1.0
+            FSharp.Core redistributables from Visual F# 4.3.0
```
Comment (dsyme) [inline src/fsharp/FSharp.Core.nuget/FSharp.Core.LatestNuget.nuspec]: This text be something like "Visual F# SDK 4.3.0" (or, if you follow the advice above, "Visual F# SDK 4.1.3").  Using the official title of the MIcrosoft SDK would be best I suppose

File: src/utils/CompilerLocationUtils.fs [hunk 5]
```diff
@@ -12,7 +12,7 @@ open System.Runtime.InteropServices
 module internal FSharpEnvironment =
 
     /// The F# version reported in the banner
-    let FSharpBannerVersion = "4.1"
+    let FSharpBannerVersion = "4.3"
```
Comment (dsyme) [inline src/utils/CompilerLocationUtils.fs]: Please revert to 4.1

File: vsintegration/src/FSharp.ProjectSystem.FSharp/VSPackage.resx [hunk 6]
```diff
@@ -468,10 +468,10 @@
     <value>Customizes the environment to maximize code editor screen space and improve the visibility of F# commands and tool windows.</value>
   </data>
   <data name="ProductDetails" xml:space="preserve">
-    <value>Microsoft Visual F# 4.1</value>
+    <value>Microsoft Visual F# 4.3</value>
```
Comment (dsyme) [inline vsintegration/src/FSharp.ProjectSystem.FSharp/VSPackage.resx]: This should be "Microsoft Visual F# Tools NNN".  Don't use 4.3

File: vsintegration/src/FSharp.ProjectSystem.FSharp/xlf/VSPackage.cs.xlf [hunk 7]
```diff
@@ -453,8 +453,8 @@
         <note />
       </trans-unit>
       <trans-unit id="9001">
-        <source>Visual F# 4.1</source>
-        <target state="translated">Visual F# 4.1</target>
+        <source>Visual F# 4.3</source>
+        <target state="needs-review-translation">Visual F# 4.1</target>
```
Comment (dsyme) [inline vsintegration/src/FSharp.ProjectSystem.FSharp/xlf/VSPackage.cs.xlf]: This is still at 4.1 for some reason. But regardless don't use 4.3

File: vsintegration/src/SupportedRuntimes/SupportedRuntimes.xml [hunk 8]
```diff
@@ -11,6 +11,7 @@
         <Assembly Version="4.3.1.0" Description="F# 3.1 (FSharp.Core, 4.3.1.0)"/>
         <Assembly Version="4.4.0.0" Description="F# 4.0 (FSharp.Core, 4.4.0.0)"/>
         <Assembly Version="4.4.1.0" Description="F# 4.1 (FSharp.Core, 4.4.1.0)"/>
+        <Assembly Version="4.4.3.0" Description="F# 4.3 (FSharp.Core, 4.4.3.0)"/>
```
Comment (dsyme) [inline vsintegration/src/SupportedRuntimes/SupportedRuntimes.xml]: This addition is wrong.  F# 4.3 doesn't exist. You can use ``F# 4.1 (FSharp.Core, 4.4.3.0)``

**Comment by dsyme** [comment]
@KevinRansom, @cartermp, @TIHan, @brettfo and myself are emailing about a revised set of version updates, we'll send a summary in due course

**Pull request merged**